ddl/table.go

```diff
@@ -63,13 +70,31 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 			// TODO: Add restrictions to this operation.
 			go splitTableRegion(d.store, tbInfo.ID)
 		}
-		// Finish this job.
-		job.FinishTableJob(model.JobStateDone, model.StatePublic, ver, tbInfo)
-		asyncNotifyEvent(d, &util.Event{Tp: model.ActionCreateTable, TableInfo: tbInfo})
-		return ver, nil
+	case model.StateWriteReorganization:
+		// reorganization -> public (insert data before we make the table public)
+		err = doCreateTableInsert(d, t, job, tbInfo, snapshotTS)
+		if err != nil {
+			job.State = model.JobStateCancelled
```

Comment:
Be careful to set job.State to model.JobStateCancelled, if the state is model.JobStateCancelled, it will not remove the tableInfo from inforschema and finish this job directly. Please take a look the function `convertJob2RollbackJob`, maybe you can set the job.State to `model.JobStateCancelling`

Reply:
This PR can maybe help you https://github.com/pingcap/tidb/pull/8537

Reply:
1. for now tableinfo is not added to infoschema until data insertion is done, so I think infoschema should not be the concern.
2. when the job.State is set to `JobStateCancelled`, I did the cleanup in `finishDDLJob`, which is a bit weird, and I will try to make it by setting `JobStateCancelling`. Thank you!